frontend/src/components/StreamingBox.vue

```diff
@@ -199,78 +212,89 @@ export default {
         return 720
       }
     },
-    taggedSrc() {
-      return get(this.printer, 'pic.img_url', this.printerStockImgSrc)
+    webcamSnapshot() {
+      return this.printer?.pic?.img_url
     },
 
-    // streaming timeline
+    // Free user streaming
     isBasicStreamingInProgress() {
-      return this.remainingSecondsCurrentVideoCycle > 0 && this.remainingSecondsCurrentVideoCycle < 30
+      return !this.autoplay && this.currentSession === STREAM_SESSION.Active
     },
     isBasicStreamingReadyToPlay() {
-      return !this.isVideoVisible && (this.remainingSecondsCurrentVideoCycle == 30 || this.remainingSecondsUntilNextCycle == 0)
+      return !this.autoplay && this.streamStatus === STREAM_STATUS.Available && !this.isTrackMuted && !this.isVideoLoading &&
+        this.currentSession === STREAM_SESSION.Idle
     },
     isBasicStreamingFrozen() {
-      return this.remainingSecondsUntilNextCycle > 0 && !this.isVideoVisible
+      return !this.autoplay && this.streamStatus === STREAM_STATUS.Available && this.currentSession === STREAM_SESSION.Frozen
     },
   },
 
-  methods: {
-    onCanPlay() {
-      this.videoLoading = false
-      if (!this.autoplay) {
-        this.videoLimit.startOrResumeVideoCycle()
+  watch: {
+    remainingSeconds(newSeconds, _) {
+      this.currentSession = StreamThrottle.currentSession(newSeconds);
+      const [active, frozen] = StreamThrottle.calculateSessionsSeconds(newSeconds);
+      this.activeSessionRemainingSeconds = active;
+      this.frozenSessionRemainingSeconds = frozen;
+
+      const whatHappened = StreamThrottle.whatHappened(newSeconds);
+      if (whatHappened === STREAM_EVENT.FrozenSessionStarted) {
+        this.webrtc.stopStream();
+      } else if (whatHappened === STREAM_EVENT.CycleFinished) {
+        clearInterval(this.basicStreamTimerId);
+        this.remainingSeconds = StreamThrottle.defaultRemainingSeconds();
+        this.streamThrottle.removeFromStorage();
       }
```

Comment:
Probably not the best way to handle basic streaming timer but it makes mobile app and web app share exact same logic